src/app/trainee/revalidation-history/revalidation-history.component.html

```diff
@@ -1,81 +1,108 @@
                         </div>
-                    </ng-container>
-                </div>
-            </td>
-        </ng-container>
+                        <mat-divider *ngIf="element.comments?.length > 0"></mat-divider>
+                        <ng-container *ngFor="let comment of element.comments">
+                            <div class="element-notes">
+                                {{ comment.comment }}
+                            </div>
+                        </ng-container>
+                    </div>
+                </td>
+            </ng-container>
 
-        <tr mat-header-row *matHeaderRowDef="columnsToDisplay"></tr>
-        <tr tabindex="0" mat-row *matRowDef="let element; columns: columnsToDisplay;" class="element-row"
-            [class.expanded-row]="expandedElement === element"
-            (click)="expandedElement = expandedElement === element ? null : element"
-            (keyup.enter)="expandedElement = expandedElement === element ? null : element"
-            (keyup.space)="expandedElement = expandedElement === element ? null : element">
-        </tr>
-        <tr mat-row *matRowDef="let row; columns: ['expandedDetail']" class="detail-row"></tr>
+            <tr mat-header-row *matHeaderRowDef="columnsToDisplay"></tr>
+            <tr tabindex="0" mat-row *matRowDef="let element; columns: columnsToDisplay;" class="element-row"
+                [class.expanded-row]="expandedElement === element"
+                (click)="expandedElement = expandedElement === element ? null : element"
+                (keyup.enter)="expandedElement = expandedElement === element ? null : element"
+                (keyup.space)="expandedElement = expandedElement === element ? null : element">
+            </tr>
+            <tr mat-row *matRowDef="let row; columns: ['expandedDetail']" class="detail-row"></tr>
 
-        <ng-container matColumnDef="noRecord">
-            <td colspan="5" mat-footer-cell class="centered" *matFooterCellDef>
-                <em class="icon-red">No history available.</em>
-            </td>
-        </ng-container>
+            <ng-container matColumnDef="noRecord">
+                <td colspan="5" mat-footer-cell class="centered" *matFooterCellDef>
+                    <em class="icon-red">No history available.</em>
+                </td>
+            </ng-container>
 
-        <tr mat-footer-row *matFooterRowDef="['noRecord']"
-            [ngClass]="{'hide':revalidationHistory.recommendations.length > 0}">
-        </tr>
-    </table>
-</div>
-<div class="button-row">
-    <span class="spacer"></span>
-    <button mat-raised-button (click)="openDialog()" color="primary">Recommend</button>
+            <tr mat-footer-row *matFooterRowDef="['noRecord']"
+                [ngClass]="{'hide':revalidationHistory.recommendations.length > 0}">
+            </tr>
+        </table>
+        <div class="button-row" *ngIf="(isHandset$ | async) === false">
```

Comment:
would the multiple `(isHandset$ | async) === false` create multiple subscriptions? perhaps this can be wrapped on an ng-container tag and reused with the as syntax

Reply:
It's one subscription and one variable. Could do with creating on variable. I'll revise

Reply:
i guess what i was trying to say is if `(isHandset$ | async) === false` is in multiple places in html would more than one subscription be created? im not entirely sure myself but yea a template variable sounds like a good idea as that would definitely mean just one subscription